gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
Oops, you're right, `len(self.WORKERS)` is the right initial value.

On the guard comparison, the value being printed out is `len(workers)` instead of `len(self.WORKERS)`. It looks like the reason for this is that `kill_worker` will only synchronously remove a worker from `self.WORKERS` if the worker is already dead (throwing `errno.ESRCH`)—otherwise it gets popped asynchronously in `reap_workers`. That behavior could be modified (maybe should be?) otherwise I think this should be:

```python
orig_worker_count = len(self.WORKERS)
#…
if len(workers) != orig_worker_count):
    self.log.debug("{0} workers".format(len(workers)),
    #…
```